### DISCUSSION TITLE
how to sync renamed branch from github to vscode?

### DISCUSSION THREAD

**Discussion by Hifilo** [community/community#58675] [New to GitHub] [answered]
Hello, I renamed my `build` branch to `develop` on github, but in vscode source control I now have both build & develop

![Code_BNlumIUM09](https://github.com/community/community/assets/92943726/81e215c9-dbec-4e9b-b0ed-9188cffda473)

In my case I think I'm safe to delete the build branch, but I would like to know the correct way of syncronizing the 2, because I may come across a situation where deleting the branch isn't ideal.

Thanks

**Comment by Gustavonn07** [comment, marked as answer]
Hi!! If you renamed your branch from "build" to "develop" on GitHub, but you still see both branches in Visual Studio Code source control, you might need to update your local repository information to reflect this change.

Here are the steps you can follow to properly sync branches in Visual Studio Code:

1. Open Visual Studio Code and open the Source Code Control dashboard. Make sure you are in your repository directory.

2. In the source control control panel, make sure you are in the "build" branch. Otherwise, switch to the "build" branch.

3. Run the command (into the terminal) to update the local repository information with the git pull command. This will ensure that you have the latest changes from the "build" branch in your local repository. The command would be something like:


`git pull origin build`

4. Once you have the latest changes from the "build" branch, it's time to merge that branch into the "develop" branch. Run the command (into the terminal) to merge the changes from the "build" branch into the "develop" branch. The command would be something like:

`git merge build`

5. You should now have the changes from the "build" branch merged into the "develop" branch. Check that everything is working as expected and test your code.
Once you're sure the changes were merged correctly and you no longer need the "build" branch, you can delete the branch using the command (into the terminal)  `git branch -d build` . Make sure you are in the "develop" branch when running this command.

Remember, it's always good practice to back up your repository before making significant changes to branches, to avoid accidentally losing code.

Hope it helps!

**Reply by Hifilo** [reply]
This is what I was trying to figure out. Thanks!
